### PR TITLE
feat: add variable declarations to colon spacing checks (fixes #178)

### LINKED ISSUE #178: The space-after-colon rule does not work for variable delcarations
With:
```javascript
'flowtype/space-after-type-colon': [2, 'always'],
```

This is allowed:
```javascript
const x:number = 7;
```

### PR DESCRIPTION
**Pull request by aaron-harvey**
This PR adds a VariableDeclaration hook to before / after colon spacing checks. 
Fixes issue #178.

### CODE REVIEW

File: src/rules/typeColonSpacing/evaluateVariables.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export default (context, report) => {
+  const sourceCode = context.getSourceCode();
+
+  const getColon = (node, typeAnnotation) => {
+    if (node.type === 'FunctionTypeParam') {
+      return sourceCode.getFirstToken(node, node.optional ? 2 : 1);
+    } else {
+      return sourceCode.getFirstToken(typeAnnotation);
+    }
+  };
```
Comment (danharper) [inline src/rules/typeColonSpacing/evaluateVariables.js]: Is this function left over from somewhere else? Just `sourceCode.getFirstToken(typeAnnotation)` works.

Comment (aaron-harvey) [inline src/rules/typeColonSpacing/evaluateVariables.js]: Ah yes, missed that function in my review before I pushed.

**Comment by gajus** [comment]
@aaron-harvey Is this ready to be merged?

**Comment by Saravanan21** [comment]
@gajus @aaron-harvey bump on this? Would love to have this feature!

**Comment by gajus** [comment]
Thank you

**Pull request merged**